### PR TITLE
lint: add operator lint command as separate container image

### PR DESCRIPTION
**Pull request by erkiesken**
Build and publish the lint executable as separate image, useful to have in CI pipelines to validate the CRD yaml files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,7 +91,7 @@ $(CLIENT_TARGET): $(K8S_GEN_DEPS)
 	--output-package "$(GO_PKG)/pkg/client"
 
 LISTER_TARGET := pkg/client/listers/monitoring/v1/prometheus.go
-$(LISTER_TARGET): $(K8S_GEN_DEPS)
+$(LISTER_TARGET): $(K8S_GEN_DEPS	)
```
Comment (s-urbaniak) [inline Makefile]: nit: i think the tab character is not necessary here.

**Comment by s-urbaniak** [comment]
@erkiesken thanks a lot for the contribution, this lgtm code-wise, I just would like another pair of eyes to also decide if we want another release artifact.

File: Makefile (hunk 1 above)
Comment (erkiesken) [reply Makefile]: Yes, noticed too late that I had accidentally added it for some reason. Fixed in followup commit.

**Comment by erkiesken** [comment]
Yeah I was debating myself if it would be better to include lint executable inside the main image or not. It would add an extra ~16MB binary that is mostly not needed, just in dev or CI use cases usually.

**Comment by erkiesken** [comment]
Also, would a section in readme about linting with sample bash snippet be welcome?

**Comment by brancz** [comment]
I think this is a fantastic addition! Thank you so much for doing this!

> Also, would a section in readme about linting with sample bash snippet be welcome?

Yes I think that would be amazing. Once added I think this is ready to go.

**Comment by erkiesken** [comment]
I added a user guide doc for using the linter, with shell script example that I use in my CI pipeline as well.

**Comment by brancz** [comment]
Sooooo awesome! Thank you so much!

**Comment by paulfantom** [comment]
@brancz @s-urbaniak this needs creation of `quay.io/coreos/prometheus-operator-lint` and setting up credentials. Otherwise `master` builds fail. Alternatively we can move all our tooling into one image (jsonnet-ci + lint, etc.).

**Comment by elisiano** [comment]
I know I’m late to the party but I was reading the added docs and I believe the moment the linter is being used outside of the container (ie: go get ...) then the binary name (`lint`) is a bit too generic IMO. 

Would it be worthwhile prefixing it with `prometheus-operator-` (or anything really)?

**Comment by paulfantom** [comment]
@elisiano I think that's a good idea. We should prefix it with `po-`, the same as we already did with [`po-docgen`](https://github.com/coreos/prometheus-operator/blob/a382ba54868825ff334917be4e7f1d9ab9eb0671/Makefile#L283). Would you mind creating a PR?

**Pull request merged**